src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -497,8 +504,11 @@ export const TradeInput = memo(({ isCompact }: TradeInputProps) => {
           bg='background.surface.raised.accent'
           borderBottomRadius='xl'
         >
-          <RecipientAddress />
-          <ManualAddressEntry />
+          <WithLazyMount shouldUse={Boolean(receiveAddress)} component={RecipientAddress} />
```

Comment:
Very smort to bypass the rules of hooks! Ideally we would simply have a `skip` prop in the hooks (e.g what `react-query` and similar allow as an option) but that obviously means huge refactors, or it being an initial design decision

Reply:
gonna keep this as is if that's ok, erring on the side of caution where we want the falsey case to be the safe option, i.e falsey means "skip"